temporalcli/commands.workflow.go

```diff
@@ -41,3 +95,77 @@ func (*TemporalWorkflowTraceCommand) run(*CommandContext, []string) error {
 func (*TemporalWorkflowUpdateCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
+
+func (s *SingleWorkflowOrBatchOptions) workflowExecOrBatch(
+	cctx *CommandContext,
+	namespace string,
+	cl client.Client,
+) (*common.WorkflowExecution, *workflowservice.StartBatchOperationRequest, error) {
+	// If workflow is set, we return single execution
+	if s.WorkflowId != "" {
+		if s.Query != "" {
+			return nil, nil, fmt.Errorf("cannot set query when workflow ID is set")
+		} else if s.Reason != "" {
+			return nil, nil, fmt.Errorf("cannot set reason when workflow ID is set")
+		} else if s.Yes {
+			return nil, nil, fmt.Errorf("cannot set 'yes' when workflow ID is set")
+		}
+		return &common.WorkflowExecution{WorkflowId: s.WorkflowId, RunId: s.RunId}, nil, nil
+	}
+
+	// Check query is set properly
+	if s.Query == "" {
+		return nil, nil, fmt.Errorf("must set either workflow ID or query")
+	} else if s.WorkflowId != "" {
+		return nil, nil, fmt.Errorf("cannot set workflow ID when query is set")
+	} else if s.RunId != "" {
+		return nil, nil, fmt.Errorf("cannot set run ID when query is set")
+	}
+
+	// Count the workflows that will be affected
+	count, err := cl.CountWorkflow(cctx, &workflowservice.CountWorkflowExecutionsRequest{Query: s.Query})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed counting workflows from query: %w", err)
+	}
+	yes, err := cctx.promptYes(
+		fmt.Sprintf("Start batch against approximately %v workflow(s)? y/N", count.Count), s.Yes)
+	if err != nil {
+		return nil, nil, err
+	} else if !yes {
+		// We consider this a command failure
+		return nil, nil, fmt.Errorf("user denied confirmation")
+	}
+
+	// Default the reason if not set
+	reason := s.Reason
+	if reason == "" {
+		username := "<unknown-user>"
+		if u, err := user.Current(); err != nil && u.Username != "" {
+			username = u.Username
+		}
+		reason = "Requested from CLI by " + username
+	}
```

Comment:
Current CLI requires reason to be set by user. UI defaults to something like `Terminated from Web UI by foo@bar.com`, so we follow a similar format, but we just do "Requested" instead of specific verb.